ipa-server/.applier/group_vars/seed-hosts.yml

```diff
@@ -0,0 +1,66 @@
+---
+namespace: "ipa"
+app_name: "ipa"
+base_openshift_url: example.com 
+realm: "EXAMPLE.COM"
+admin_password: "Password123"
+openshift_templates_version: "v1.4.5"
+ipa_templates_version: "5e55ee8a62e7c51bf8c1805b33425f48e2d8f44e"
+
+params:
+  IPA_SERVER_SERVICE: "{{ app_name }}"
+  IPA_SERVER_IMAGE: "freeipa-server:rhel-7"
+  IPA_SERVER_HOSTNAME: "{{ app_name }}.apps.{{ base_openshift_url }}"
+  IPA_SERVER_INSTALL_OPTS: "-U -r {{ realm }} --setup-dns --no-forwarders --no-ntp"
+  IPA_ADMIN_PASSWORD: "{{ admin_password }}"
+  SERVICE_ACCOUNT_USEROOT: "{{ app_name }}"
+  VOLUME_CAPACITY: 1G
+  TIMEOUT: 600
+
+openshift_cluster_content:
+- object: Environment Setup
+  content:
+  - name: Create Projects
+    template: "https://raw.githubusercontent.com/redhat-cop/openshift-templates/{{ openshift_templates_version }}/project-requests/create-project.yml"
+    action: create
+    params_from_vars:
+      NAMESPACE: "{{ namespace }}"
+      NAMESPACE_DISPLAY_NAME: "{{ namespace }}"
+    tags:
+      - ipa_project
+
+- object: Image Streams
+  content:
+  - name: "{{ app_name }}-streams"
+    file: "https://raw.githubusercontent.com/freeipa/freeipa-container/{{ ipa_templates_version }}/freeipa-server-openshift-image.json"
+    namespace: "{{ namespace }}"
+    tags:
+      - ipa_imagestream
+
+- object: Service Account
+  content:
+  - name: "{{ app_name }}-sa"
+    template: "https://raw.githubusercontent.com/redhat-cop/openshift-templates/{{ openshift_templates_version }}/service-accounts/generic-service-account.yml"
+    namespace: "{{ namespace }}"
+    params_from_vars:
+      SA_NAME: "{{ params.SERVICE_ACCOUNT_USEROOT }}"
+      SA_NAMESPACE: "{{ namespace }}"
+    tags:
+      - ipa_serviceaccount
+  - name: "{{ app_name }}-sa-policy"
+    template: "https://raw.githubusercontent.com/redhat-cop/openshift-templates/{{ openshift_templates_version }}/scc/project-run-anyuid-template.yml"
+    namespace: "{{ namespace }}"
+    params_from_vars:
+      NAMESPACE: "{{ namespace }}"
+      NAME: "{{ params.SERVICE_ACCOUNT_USEROOT }}"
+    tags:
+      - ipa_serviceaccount
+
+- object: Deployments
+  content:
+  - name: ipa
+    template: "https://raw.githubusercontent.com/freeipa/freeipa-container/{{ ipa_templates_version }}/freeipa-server-openshift.json"
```

Comment:
There seems to be an issue with idempotency here. When I run the applier the first time it works. After that I get an error:
```
failed: [localhost] (item={'oc_path': u''}) => {"changed": true, "cmd": "oc process  --local  -f https://raw.githubusercontent.com/freeipa/freeipa-container/504c122090713505025971cee7a4d50eb394e291/freeipa-server-openshift.json -n bkin-ipa  --param='IPA_SERVER_HOSTNAME=ipa.apps.example.com' --param='IPA_SERVER_IMAGE=freeipa-server:rhel-7' --param='VOLUME_CAPACITY=1G' --param='IPA_SERVER_INSTALL_OPTS=-U -r EXAMPLE.COM --setup-dns --no-forwarders --no-ntp' --param='SERVICE_ACCOUNT_USEROOT=ipa' --param='TIMEOUT=600' --param='IPA_ADMIN_PASSWORD=password'  --ignore-unknown-parameters | oc apply -n bkin-ipa -f - ", "delta": "0:00:00.819589", "end": "2019-06-25 11:30:32.229980", "failed_when_result": true, "msg": "non-zero return code", "oc_param_file_item": {"oc_path": ""}, "rc": 1, "start": "2019-06-25 11:30:31.410391", "stderr": "Error from server (Invalid): Service \"freeipa-server\" is invalid: spec.clusterIP: Invalid value: \"\": field is immutable\nError from server: unrecognized type: int64", "stderr_lines": ["Error from server (Invalid): Service \"freeipa-server\" is invalid: spec.clusterIP: Invalid value: \"\": field is immutable", "Error from server: unrecognized type: int64"], "stdout": "route.route.openshift.io/freeipa-server-https unchanged\nsecret/freeipa-server-password configured\npersistentvolumeclaim/freeipa-server unchanged", "stdout_lines": ["route.route.openshift.io/freeipa-server-https unchanged", "secret/freeipa-server-password configured", "persistentvolumeclaim/freeipa-server unchanged"]}
```

Reply:
Yepp, I've noticed that as well. Trying to see if I can get around that without having to maintain our own version of that template.